### DISCUSSION TITLE
Genetic Algorithm in HERON

### DISCUSSION THREAD

**Discussion by aalfonsi** [idaholab/HERON#283] [General] [answered]
Does HERON support (or will soon support) the Genetic Algorithm for its optimization search?

**Comment by PaulTalbot-INL** [comment, marked as answer]
It would be pretty simple to add that as a template if there was a use case for it. We've looked into it before, and it appears it would take far more samples than the gradient descent search for most optimization efforts.

However, we have plans in the next few months to deploy Bayesian optimization via ROM and maximizing information criteria, which initial tests show could require even fewer sample evaluations to find a reasonable optimal point.

**Comment by PaulTalbot-INL** [comment]
That said, if you have a use case for it, we're interesting in hearing!
